### PR TITLE
#5 Ch project initialization 

### LINKED ISSUE #5: Project Initialization
**Description:**

This issue outlines the initial setup for our new Node.js project, including the creation of a tsconfig.json file for TypeScript configuration, the installation of basic dependencies such as Express.js, and the development of a simple test API endpoint.

**Tasks:**

- Initialize a new Node.js project: Use npm init or yarn init to create a new Node.js project. This will generate a package.json file that will track our project's dependencies and scripts.

- Set up TypeScript configuration: Create a tsconfig.json file to configure TypeScript for our project. This file will include settings for the TypeScript compiler, such as the target JavaScript version and module resolution strategy.

- Install required basic dependencies: Install Express.js and any other necessary dependencies for our project. Express.js will be used to create our API endpoints.

- Create a test API endpoint: Develop a simple /status API endpoint that returns a JSON response. This endpoint will serve as a basic test to ensure our server is running correctly and can handle requests.

**Acceptance Criteria:**

- [ ]  A new Node.js project is initialized with npm init or yarn init, and a package.json file is generated.
- [ ]  A tsconfig.json file is created with appropriate TypeScript configuration.
- [ ]  Express.js and any other required dependencies are installed.
- [ ]  A /status API endpoint is created that returns a JSON response.

**Dev Notes:**

-  TypeScript configuration in tsconfig.json is compatible with our project's requirements.
-  The /status endpoint should return a simple JSON object, such as {"status": "ok"}, to confirm the server is running.**

### PR DESCRIPTION
**Pull request by ambroisegithub**
**What does this PR do?**

-  This pull request addresses the initial setup for a new Node.js project, including tasks such as initializing the project, setting up TypeScript configuration, installing necessary dependencies, and creating a test API endpoint.

**Description of Task to be completed:**

-  All about Project setting

**The tasks included in this pull request are as follows:**

**Initialize a new Node.js project:** 

-  Use either npm init or yarn init to initialize the project, generating a package.json file to track dependencies and scripts.

-  Set up TypeScript configuration: Create a tsconfig.json file to configure TypeScript for the project, specifying settings like the target JavaScript version and module resolution strategy.

-  Install required basic dependencies: Install essential dependencies, such as Express.js, for the project.

-  Create a test API endpoint: Develop a simple /status API endpoint that returns a JSON response, serving as a basic test for the server's functionality and request handling.

-  How should this be manually tested?

**To test this pull request manually, you can follow these steps:**

-  Clone the repository: git clone https://github.com/atlp-rwanda/dynamites-ecomm-be.git

-  Navigate to the project directory: cd dynamites-ecomm-be

-  Install dependencies: npm install

-  In your terminal,cmd,etc Run : npm run build , to generate index.js

-  In your terminal,cmd,etc Run : npm run start,  to start your project 

-  Test the /status API endpoint using a browser or Postman with a GET request to http://localhost:3000/.

**Labels:**

-  backend, chore

### CODE REVIEW

**Review by wayneleon1 — COMMENTED** [review]
Hello @ambroisegithub everything is good but  you have to be descriptive  on your PL description , Try to use proper PL description format

**Review by wayneleon1 — APPROVED** [review]

**Review by jkarenzi — APPROVED** [review]

**Review by jkarenzi — COMMENTED** [review]
Hello @ambroisegithub , the setup is all good but commit messages should adhere to proper format

**Comment by ambroisegithub** [comment]
@wayneleon1  and @jkarenzi can you check me again I was correcting  it!!!

File: index.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import bodyParser from "body-parser";
```
Comment (Habinezajanvier) [inline index.ts]: We no longer need body-parser as it's now included in express

File: package.json [hunk 2]
```diff
@@ -0,0 +1,38 @@
+
+{
+    "name": "dynamites-ecomm-be",
+    "version": "1.0.0",
+    "description": "",
+    "main": "index.js",
+    "scripts": {
+      "start:dev": "nodemon ./index.ts"
+    },
```
Comment (Habinezajanvier) [inline package.json]: Please add `build` script and normal `start` which doesn't use nodemon

File: tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+    // "noPropertyAccessFromIndexSignature": true,       /* Enforces using indexed accessors for keys declared using an indexed type. */
+    // "allowUnusedLabels": true,                        /* Disable error reporting for unused labels. */
+    // "allowUnreachableCode": true,                     /* Disable error reporting for unreachable code. */
+
+    /* Completeness */
+    // "skipDefaultLibCheck": true,                      /* Skip type checking .d.ts files that are included with TypeScript. */
+    "skipLibCheck": true                                 /* Skip type checking all .d.ts files. */
+  },
+  "include": ["src/**/*.ts", "app.ts", "index.ts"],
+  "exclude": ["node_modules"]
```
Comment (Habinezajanvier) [inline tsconfig.json]: Please remove those comments, if codes are not being used, please remove them

**Review by Habinezajanvier — CHANGES_REQUESTED** [review]

**Review by Dawaic6 — APPROVED** [review]

File: index.ts (hunk 1 above)
Comment (ambroisegithub) [reply index.ts]: I was  removed it.

File: package.json (hunk 2 above)
Comment (ambroisegithub) [reply package.json]: I  add both  build  and normal start  also  I use node instead  nodemon.

File: tsconfig.json (hunk 3 above)
Comment (ambroisegithub) [reply tsconfig.json]: I  removed them.

**Review by Habinezajanvier — APPROVED** [review]
LGTM

**Pull request merged**